examples/HistogramBitmap/HistogramBitmap.cpp

```diff
@@ -0,0 +1,234 @@
+#include <noiz/noise2.hpp>
+
+
+#include <charconv>
+#include <filesystem>
+#include <format>
+#include <iostream>
+#include <vector>
+#include <fstream>
+
+	struct BmpHeader {
+		char bitmapSignatureBytes[2] = {'B', 'M'};
+		uint32_t sizeOfBitmapFile = 54;
+		uint32_t reservedBytes = 0;
+		uint32_t pixelDataOffset = 54;
+
+		BmpHeader(uint16_t imageSize) {
+			sizeOfBitmapFile += imageSize * imageSize * 3; //rgb, 1 byte per color
+		}
+		void writeToFile(std::ofstream& outFile){
+			//writing like this to avoid padding issues
+			outFile.write(bitmapSignatureBytes, 2);
+			outFile.write((char*) &sizeOfBitmapFile, sizeof(uint32_t));
+			outFile.write((char*) &reservedBytes, sizeof(uint32_t));
+			outFile.write((char*) &pixelDataOffset, sizeof(uint32_t));
+		}
+	};
+
+
+	struct BmpInfoHeader {
+		uint32_t sizeOfThisHeader = 40;
+		int32_t width = 256; // in pixels
+		int32_t height = 256;
+		uint16_t numberOfColorPlanes = 1; // must be 1
+		uint16_t colorDepth = 24;
+		uint32_t compressionMethod = 0;
+		uint32_t rawBitmapDataSize = 0;
+		int32_t horizontalResolution = 3780; // in pixel per meter //do i even need this?
+		int32_t verticalResolution = 3780; // in pixel per meter
+		uint32_t colorTableEntries = 0;
+		uint32_t importantColors = 0;
+
+		BmpInfoHeader(uint16_t imageSize){
+			width = imageSize;
+			height = imageSize;
+		}
```

Comment:
Use member list initialization instead of assignment in the constructor body.

```suggestion
		BmpInfoHeader(uint16_t const imageSize) : width(imageSize), height(imageSize)
```